### ISSUE TITLE
Day01 Ex03 Examples result are maybe incorrect

### ISSUE THREAD

**Issue by Paul-JULLIEN** [42-AI/bootcamp_machine-learning#64]
* Day: 01
* Exercise: 03

I think the example result of gradient computation may be false.

If we compute the hypothesis with theta = [[2], [0.7]]
x = [[12.4956442], [21.5007972], [31.5527382], [48.9145838], [57.5088733]]
we get y_hat = [[10.74695094], [17.05055804], [24.08691674], [36.24020866],[42.25621131]]
In comparison with y = [[37.4013816], [36.1473236], [45.7655287], [46.6793434], [59.5585554]]
We can see that if we compute the substract of hyp and y, every components will be negatives. So if we sum them and take the mean, it must be a negative number. However, in the examples the result is positive and greater (i.e in abs value) than the result get by hand

The problem seems to be the same for the second components of the gradient and for the second examples.

For this one, I have no idea why our result are so different and the mistake clearly can be mine, so if it is, I will be very interested by your explanation of what I missed. :)

**Comment by FlorianMuller** [comment]
I figured out the same thing, if I understand gradient correctly, it should be negative values.

Personally I found those values
```Python
# Example 1 output:
me      = array([ -19.0342574 , -586.66875564])
subject = array([  21.0342574,   587.36875564])


# Example 2 output:
me      = array([  -57.86823748, -2230.12297889])
subject = array([   58.86823748,  2229.72297889])
```

I noticed that my values and the subject's one have some part in common (especially the decimal part), don't know what that mean 🤷🏻‍♂️
(And I'm not sure my values are right, maybe I've made mistake somewhere 🕵🏻‍♂️)

**Comment by madvid** [comment]
Indeed, there are errors in the differents examples mentioned. I will take care of it, thank you for noticed it in the issue.

**Comment by madvid** [comment]
Thanks for the issue @Paul-JULLIEN , indeed with a simple sign verification it appears there is an incoherence.
After a verification, I get the same result as you @FlorianMuller ,
The issue concerned the exercise 3 and 4 of module06 (day04). It  is fixed in the commit [dev e5fe5db]. Changes will be available on main branch in the next merge from dev.
